### ISSUE TITLE
DfMon Isolated is incompatible with ASP.NET Core Integration

### ISSUE THREAD

**Issue by mdddev** [microsoft/DurableFunctionsMonitor#161]
Hello,

I'm using dfMon in injected mode in an isolated function app. If have disabled auth for testing purposes. When opening `http://localhost:7071/api/durable-functions-monitor` in a browser, the screen stays blank. In the Edge DevTools, an error is shown `Manifest: Line: 1, column: 1, Syntax error.` The manifest seems empty.

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/6166d7db-53a5-400f-bf8e-d46abc86f1e9)

Here are my packages I'm using.

```xml
  <ItemGroup>
    <PackageReference Include="DurableFunctionsMonitor.DotNetIsolated" Version="6.3.0-beta6" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.21.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore" Version="1.2.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.17.0" OutputItemType="Analyzer" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.1.1" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.1.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Timer" Version="4.3.0" />
  </ItemGroup>
```

**Comment by scale-tone** [comment]
@mdddev , can you share Network tab there in your browser's DevTools, and also your `host.json` file?

**Comment by mdddev** [comment]
PS, looking through the DevTools I've found this, though Javascript is _not_ disabled on that site.

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/0e4bf20d-f4e9-44c6-a972-6f465939c449)

**Comment by scale-tone** [comment]
Right, so it appears that DfMon Isolated is incompatible with [ASP.NET Core Integration](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide?tabs=windows#aspnet-core-integration) (which you're using in your project, as `Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore` reference indicates).

OK, will need to check what can be done about it...

Until then you're welcome to use DfMon in [all other supported forms](https://github.com/microsoft/DurableFunctionsMonitor/wiki#overview).

**Comment by mronnblom-ith** [comment]
So what are you saying? DfMon Isolated in injected mode isn't usable at all?

**Comment by scale-tone** [comment]
@mronnblom-ith , if your Durable Functions project uses ASP.Net Core Integration, then you are still welcome to use [all other supported ways of running DfMon](https://github.com/microsoft/DurableFunctionsMonitor/wiki#overview) to monitor it.

**Comment by scale-tone** [comment]
Hi @mdddev , I've just pushed [version 6.5.0-beta2](https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetIsolated/6.5.0-beta2), which should work with ASP.Net Core Integration.

In an ASP.NET Core project DfMon needs to be initialized as usual:
```
var host = new HostBuilder()
    .ConfigureFunctionsWebApplication((builderContext, workerAppBuilder) => {

        workerAppBuilder.UseDurableFunctionsMonitor(......);

        //......
    })
    .ConfigureServices(services => {
         //....
    })
    .Build();

```

Note that this new package targets .NET 8.

May I ask you to try it out and see if there're any issues? Thanks!

**Comment by mdddev** [comment]
Hi, I can confirm, that even if deployed in Azure, I am getting this error on pageload.. if I remove the ASP.NET integration and re-deploy, it works without the 401 error.

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/008f8683-9efe-4771-985a-3536984917ed)
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/899d5419-4e06-414a-a130-4eec9a39523e)
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/aa75d7ea-522c-473d-b37f-e3dc98c48090)

**Comment by mdddev** [comment]
PS:  Even without the ASP.NET integration I am experiencing an error. When I enter an orchestration the history cannot be displayed, due to an 401 error. The failures seem to originate in `DfmGetOrchestrationHistoryFunction`. If I search this repo for [DFM failed](https://github.com/search?q=repo%3Amicrosoft%2FDurableFunctionsMonitor%20%22DFM%20failed%22&type=code) it seems to be indeed auth related. I am using EasyAuth in an unchanged configuration I have set-up aover a year ago, and which has been running ever since correctly in the in-proc model. I have now migrated over to the isolated model.

One further info, as per [this](https://github.com/microsoft/DurableFunctionsMonitor/tree/main/durablefunctionsmonitor.dotnetisolated.core) I have created a custom endpoint in order to expose the Dfmon at the root of the site.. Though I have changed `/{p1?}/{p2?}/{p3?}` to `{p1?}/{p2?}/{p3?}` (without the leading /). I have jsut now also tried _without_ the custom endpoint and I see the same error, 401 in the UI.

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/adf0af0e-3659-4173-8cc0-69212af235ab)

**Comment by scale-tone** [comment]
@mdddev , the "No access token provided" error means that DfMon thinks it is under the [client-directed login flow](https://github.com/microsoft/DurableFunctionsMonitor/wiki/How-to-configure-authentication#configuring-dfmon-for-client-directed-login-flow), and yet for some reason doesn't get a token from the client side.

Generally speaking, to make DfMon (no matter if it is standalone or injected) work, you are required to configure the authentication, [as per documented here](https://github.com/microsoft/DurableFunctionsMonitor/wiki/How-to-configure-authentication). By default, your DfMon endpoint would always reject unauthenticated calls to it, for obvious security reasons.

> I am using EasyAuth in an unchanged configuration

Can you show your Function App instance's "Authentication" tab, for me to better understand what that "unchanged" configuration is? There shouldn't be any secrets there, but still do ensure that you don't share any.

Also, do you have any other HTTP endpoints in your project, and if yes, how are those authenticated?

**Comment by mdddev** [comment]
Hi @scale-tone , of course, here you go. EasyAuth was not changed, and it used to work fine in the injected/in-process version of the app deployment. 

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/d79ffb86-2ada-4159-a0c3-34d2d38cc81a)
![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/33447232/e1264ee7-33a3-45f7-bae0-f52bc7cc01d5)

This is my Program.cs

```csharp
public class Program
{
    public static async Task Main()
    {
        var host = new HostBuilder()
            // .ConfigureFunctionsWebApplication((IFunctionsWorkerApplicationBuilder workerApplication) =>
            .ConfigureFunctionsWorkerDefaults((IFunctionsWorkerApplicationBuilder workerApplication) =>
            {
                 workerApplication.UseDurableFunctionsMonitor((settings, extPts) =>
                 {
                    settings.AllowedReadOnlyAppRoles = [AppRoles.READ];
                    settings.AllowedAppRoles = [AppRoles.WRITE];
                 });
            })
            .ConfigureAppConfiguration((HostBuilderContext ctx, IConfigurationBuilder builder) => 
            {
                builder
                    .SetBasePath(Directory.GetCurrentDirectory())
                    .AddJsonFile("local.settings.json", optional: true, reloadOnChange: true)
                    .AddEnvironmentVariables();
            })
            .ConfigureServices((HostBuilderContext ctx, IServiceCollection services) => 
            {
            })
            .Build();

        await host.RunAsync();
    }
}

public static class AppRoles
{
    public const string READ = "Workflows.Read";
    public const string WRITE = "Workflows.Write";
}
```

**Comment by scale-tone** [comment]
Thanks for all that info, @mdddev .

I can confirm that there is still an issue in this "Isolated+Asp.Net Core Integration" beta package, that prevents the [server-directed login flow](https://github.com/microsoft/DurableFunctionsMonitor/wiki/How-to-configure-authentication#configuring-dfmon-for-server-directed-login-flow) (the one your current setup is using) to work. More specifically, [the HttpRequestData.Identities property seems to be empty or not populated correctly](https://github.com/microsoft/DurableFunctionsMonitor/blob/bb934e7120a3482c60df541ecb2c0197be5d0526/durablefunctionsmonitor.dotnetisolated.core/Common/Auth.cs#L121).

The workaround is to switch to [client-directed flow](https://github.com/microsoft/DurableFunctionsMonitor/wiki/How-to-configure-authentication#configuring-dfmon-for-client-directed-login-flow), by changing your EasyAuth settings to look like this:

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/5447190/08ed3eff-0620-4cae-bbb5-bbff900e111e)

----

If you then get an auth error like the following:

![image](https://github.com/microsoft/DurableFunctionsMonitor/assets/5447190/fa5d13d0-be1d-4e63-b852-a7d07e473477)

, then adjust the Redirect URIs settings in your Entra Id app registration accordingly.

Please, let me know if the workaround works or doesn't.

**Comment by scale-tone** [comment]
Should be fixed in [v6.5](https://github.com/microsoft/DurableFunctionsMonitor/releases/tag/v6.5)
